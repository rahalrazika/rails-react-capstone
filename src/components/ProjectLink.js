/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import port from '../assets/por.png';

const ProjectLink = (props) => {
  const { projectData, id } = props;
  const { name, price } = projectData;

  return (
    <Link
      className="hover:link hover:cursor-pointer"
      to={{
        pathname: `/project/${id}`,
        state: { data: projectData },
      }}
    >
      <div className="relative h-screen flex">
        <div className="container flex justify-center">
          <div className="w-full mt-20 px-7">
            <div className="bg-white relative shadow-lg hover:shadow-xl transition duration-500 rounded-lg">
              <img className="rounded-t-lg" src={port} alt="" />
              <div className="py-6 px-8 rounded-lg bg-white">
                <div>
                  <h1 className="  text-gray-700 font-bold text-2xl mb-3 capitalize rounded transition duration-100  ">
                    {name}
                  </h1>
                </div>

                <div className="mt-6 py-2 px-4 text-gray-800 font-bold ">
                  <span className="text-md">${price}</span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
};
ProjectLink.propTypes = {
  id: PropTypes.number.isRequired,
  projectData: PropTypes.instanceOf(Object),
};

ProjectLink.defaultProps = {
  projectData: {},
};

export default ProjectLink;
