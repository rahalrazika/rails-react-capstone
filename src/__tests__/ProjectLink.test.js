import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import store from '../redux/store';
import ProjectLink from '../components/ProjectLink';

it('renders ProjectLink component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={['project/8']}>
        <ProjectLink />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
