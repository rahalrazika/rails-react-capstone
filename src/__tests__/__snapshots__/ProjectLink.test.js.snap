// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProjectLink component correctly 1`] = `
<a
  className="hover:link hover:cursor-pointer"
  href="/project/undefined"
  onClick={[Function]}
>
  <div
    className="relative h-screen flex"
  >
    <div
      className="container flex justify-center"
    >
      <div
        className="w-full mt-20 px-7"
      >
        <div
          className="bg-white relative shadow-lg hover:shadow-xl transition duration-500 rounded-lg"
        >
          <img
            alt=""
            className="rounded-t-lg"
            src={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "getCacheKeyAsync": [Function],
                "process": [Function],
                "processAsync": [Function],
              }
            }
          />
          <div
            className="py-6 px-8 rounded-lg bg-white"
          >
            <div>
              <h1
                className="  text-gray-700 font-bold text-2xl mb-3 capitalize rounded transition duration-100  "
              />
            </div>
            <div
              className="mt-6 py-2 px-4 text-gray-800 font-bold "
            >
              <span
                className="text-md"
              >
                $
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
